contracts/bridge/contracts/common/message/OutboundLane.sol

```diff
@@ -0,0 +1,322 @@
+
+    uint256 public confirmationFee = 0.1 ether; // how to set confirmation_fee
+
+    // MessageKey => MessageData
+    mapping(uint256 => MessageData) public messages;
+
+    /**
+     * @notice Deploys the OutboundLane contract
+     * @param _lightClientBridge The contract address of on-chain light client
+     * @param _thisChainPosition The thisChainPosition of outbound lane
+     * @param _bridgedChainPosition The bridgedChainPosition of outbound lane
+     * @param _lanePosition The lanePosition of outbound lane
+     * @param _oldest_unpruned_nonce The oldest_unpruned_nonce of outbound lane
+     * @param _latest_received_nonce The latest_received_nonce of outbound lane
+     * @param _latest_generated_nonce The latest_generated_nonce of outbound lane
+     */
+    constructor(
+        address _lightClientBridge,
+        uint256 _thisChainPosition,
+        uint256 _bridgedChainPosition,
+        uint256 _lanePosition,
+        uint256 _oldest_unpruned_nonce,
+        uint256 _latest_received_nonce,
+        uint256 _latest_generated_nonce
+    ) public MessageVerifier(_lightClientBridge, _thisChainPosition, _bridgedChainPosition, _lanePosition) {
+        outboundLaneNonce = OutboundLaneNonce(_oldest_unpruned_nonce, _latest_received_nonce, _latest_generated_nonce);
+        _setupRole(DEFAULT_ADMIN_ROLE, msg.sender);
+    }
+
+    // Send message over lane.
+    function send_message(address targetContract, bytes calldata encoded) external payable override returns (uint256) {
+        require(hasRole(OUTBOUND_ROLE, msg.sender), "Lane: NotAuthorized");
+        require(outboundLaneNonce.latest_generated_nonce - outboundLaneNonce.latest_received_nonce <= MAX_PENDING_MESSAGES, "Lane: TooManyPendingMessages");
+        require(outboundLaneNonce.latest_generated_nonce < uint128(-1), "Lane: Overflow");
+        uint256 nonce = outboundLaneNonce.latest_generated_nonce + 1;
+        uint256 fee = msg.value;
+        outboundLaneNonce.latest_generated_nonce = nonce;
+        MessagePayload memory messagePayload = MessagePayload({
+            sourceAccount: msg.sender,
+            targetContract: targetContract,
+            encoded: encoded
+        });
+        uint256 key = encodeMessageKey(nonce);
+        // finally, save messageData in outbound storage and emit `MessageAccepted` event
+        messages[key] = MessageData({
+            payload: messagePayload,
+            fee: fee  // a lowest fee may be required and how to set it
+        });
+        // TODO:: callback `on_messages_accepted`
+
+        // message sender prune at most `MAX_PRUNE_MESSAGES_ATONCE` messages
+        prune_messages(MAX_PRUNE_MESSAGES_ATONCE);
+        commit();
+        emit MessageAccepted(bridgedChainPosition, lanePosition, nonce);
+        return nonce;
+    }
+
+    function encodeMessageKey(uint256 nonce) public view returns (uint256 key) {
+        key = (bridgedChainPosition << 192) + (lanePosition << 128) + nonce;
```

Comment:
request documents.

What the max value of nonce?

Reply:
```
    // 32 bytes to identify an unique message
    // MessageKey encoding:
    // BridgedChainPosition | LanePosition | Nonce
    // [0..8)   bytes ---- BridgedChainPosition
    // [8..16)  bytes ---- LanePosition
    // [16..32) bytes ---- Nonce, max of nonce is `uint128(-1)`
```